search/search_integration_tests/downloader_search_test.cpp

```diff
@@ -0,0 +1,231 @@
+#include "testing/testing.hpp"
+
+#include "generator/generator_tests_support/test_feature.hpp"
+
+#include "search/downloader_search_callback.hpp"
+#include "search/mode.hpp"
+#include "search/result.hpp"
+#include "search/search_integration_tests/helpers.hpp"
+#include "search/search_tests_support/test_results_matching.hpp"
+#include "search/search_tests_support/test_search_request.hpp"
+
+#include "storage/downloader_search_params.hpp"
+#include "storage/http_map_files_downloader.hpp"
+#include "storage/storage.hpp"
+
+#include "geometry/rect2d.hpp"
+
+#include "base/logging.hpp"
+#include "base/macros.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/string.hpp"
+
+using namespace generator::tests_support;
+using namespace search::tests_support;
+
+using TRules = vector<shared_ptr<MatchingRule>>;
+
+namespace storage
```

Comment:
Why? Just put this class to the anonymous namespace. No need to extend storage namespace.